msbuild/Xamarin.MacDev.Tasks.Core/Tasks/FindBindingResourcePackagesBase.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.MacDev.Tasks {
+	public abstract class FindBindingResourcePackagesBase: Task {
+		public string SessionId { get; set; }
+		
+		[Required]		
+		public ITaskItem[] References { get; set; }
+		
+		[Required]		
+		public ITaskItem[] ReferencesFoundInSolution { get; set; }
+		
+		[Output]
+		public ITaskItem[] AdditionalNativeReferences { get; set; }
+
+		[Output]
+		public ITaskItem[] ManifestFound { get; set; }
+		
+		public override bool Execute ()
+		{
+			var ReferencesAlreadyResolved = new HashSet <string> (ReferencesFoundInSolution.Select (x => Path.GetFileName (x.ItemSpec)));
+
+			var additionalNativeReferences = new List<ITaskItem> ();
+			foreach (var reference in References) {
+				string referenceFile = Path.GetFileName (reference.ItemSpec);
+				if (ReferencesAlreadyResolved.Contains (referenceFile)) {
+					Log.LogMessage (MessageImportance.Low, $"Using in-solution reference for {referenceFile} instead of potential bundle.");
+					continue;
+				}
+
+				string referencePath = Path.GetDirectoryName (reference.ItemSpec);
+				string resourceBundlePath = Path.Combine (referencePath, Path.GetFileNameWithoutExtension (reference.ItemSpec) + ".resources");
+				if (Directory.Exists (resourceBundlePath)) {
+					string manifestPath = Path.Combine (resourceBundlePath, "manifest");
+					if (File.Exists (manifestPath)) {
+						Log.LogMessage ($"Found resource bundle at {resourceBundlePath}");
+
+						ManifestFound = new ITaskItem [] { new TaskItem ("ManifestFile") { ItemSpec = manifestPath } };
+						additionalNativeReferences.AddRange (ReadManifestAndCreateReferences (reference.ItemSpec, resourceBundlePath, manifestPath));
+					}
+				}
+			}
+			AdditionalNativeReferences = additionalNativeReferences.ToArray ();
+			return true;
+		}
+
+		IEnumerable<ITaskItem> ReadManifestAndCreateReferences (string referencePath, string resourceBundlePath, string manifestPath)
+		{
+			XmlDocument document = new XmlDocument ();
+			document.Load (manifestPath);
```

Comment:
This can throw `XmlException` - is that OK?

Reply:
I think so, msbuild will catch it and fail the build with task threw exception foo.

I could catch it and throw a more specific error, but unless users screw with the manifest, it should be well formed.